src/sprites/Renderer.cpp

```diff
@@ -0,0 +1,154 @@
+#include "../sprites/Renderer.h"
+#include "../view/Window.h"
+#include "../log/log.h"
+
+/* DEVELOPER: Michael Goll
+** DESIGNER:  Michael Goll
+** DATE:      March 14, 2017
+*/
+
+Renderer * Renderer::rInstance     = nullptr;
+SDL_Renderer * Renderer::_renderer = nullptr;
+SDL_Window * Renderer::_window     = nullptr;
+
+std::array<SDL_Texture *, TOTAL_SPRITES> Renderer::_sprites = {0};
+int Renderer::_index = 0;
+
+Renderer::~Renderer() {
+    for (const auto& s : _sprites) {
+        if (s != nullptr) {
+            SDL_DestroyTexture(s);
+        }
+    }
+
+    SDL_DestroyRenderer(_renderer);
+    SDL_DestroyWindow(_window);
+    free(rInstance);
+}
+
+//returns the instance if it exists, otherwise creates one
+Renderer * Renderer::instance() {
+    if (!rInstance) {
+        rInstance = new Renderer;
```

Comment:
Use of new is forbidden
For singletons, please check AudioManager and GameManager for implementation using a static member
Don't manually allocate memory